modules/shortcodes/inline-pdfs.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+/**
+ * Inline PDFs
+ * Takes one-line embeds of a PDF URL (*.pdf), and attempts to embed it directly in the post instead of leaving it as a link.
+ */
+wp_embed_register_handler( 'inline-pdfs', '#https?://[^<]*\.pdf$#i', 'inline_pdf_embed_handler' );
+
+function inline_pdf_embed_handler( $matches, $attr, $url ) {
+	return sprintf(
```

Comment:
It may be useful to add an AMP fallback since `object` is not allowed on AMP views. Something like this?

```php
if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
	return sprintf(
		'<p><a href="%1$s">%2$s</a></p>',
		esc_url( $url ),
		esc_html__( 'PDF Document', 'jetpack' )
	);
} else {
	return sprintf(
		'<object data="%1$s" type="application/pdf" width="100%%" height="800">
			<p><a href="%1$s">%1$s</a></p>
		</object>',
		esc_attr( $url )
	);
}
```

Reply:
Leaving out AMP support for now.

Reply:
Tracking this in #15192.

Reply:
Fix in #15196.